test/lib/foreman/model/auth_source_name_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+class AuthSourceNameTest < ActiveSupport::TestCase
+    def setup
+    User.current = users(:admin)
+    end
+
+    test "auth_source should not allow 'Internal' or 'Hidden' as names for ldap and external auths" do
```

Comment:
Indentation is off, also these seem to be different tests, maybe you want to break this up? `.create` calls the DB so it'd slow down the tests a bit, calling `.build` then `.valid?` would work too and it all would be done in memory (faster). :smile: